common/src/KokkosFFT_transpose.hpp

```diff
@@ -86,6 +86,31 @@ auto get_map_axes(const ViewType& view, int axis) {
   return get_map_axes(view, axis_type<1>({axis}));
 }
 
+template <class InViewType, class OutViewType, std::size_t DIMS>
+void _prep_transpose_view(InViewType& in, OutViewType& out,
+                          axis_type<DIMS> _map) {
```

Comment:
Could you please add an unit-test for this function, for example in common/unit_test/Test_Transpose.cpp. An exception test for insufficient memory case would be quite helpful.

Reply:
Added in Test_prep_transpose_view.cpp

This tests allocating/no allocating cases for managed views, and no reshape/reshape success/reshape failure for unmanaged views.